### ISSUE TITLE
Don't default to gzip level 9

### ISSUE THREAD

**Issue by seibert** [dask/fastparquet#344]
Wondering why GZIP compression in fastparquet was so slow, I noticed that the gzip compression level is defaulted to level 9:

https://github.com/dask/fastparquet/blob/master/fastparquet/compression.py#L16-L39

Gzip compression is extremely speed inefficient at level 9, and defaulting to this makes gzip very impractical.  A quick example from some parquet data I'm writing ("create" includes some constant time spent loading the data in memory + time for fastparquet to write):

* gzip 1: create time = 64.46 sec, output size = 69 MB
* gzip 6: create time =  73.94 sec, output size = 59 MB
* gzip 9: create time =  229.48 sec, output size = 58 MB

The extra 2.5 minutes spent going from level 6 to level 9 is basically wasted in this case.  Setting the default compression level to 6 (matching the gzip command line utility) is probably the best option.

**Comment by martindurant** [comment]
gzip 9 is the standard used by the builtin python library, that's where this comes from. It may be reasonable for people to expect that this default is respected. Do you know what level of compression is typically used by other parquet-writing frameworks? (if any even use gzip rather than snappy...)
Note that fastparquet also supports lz4 and lzo.

**Comment by seibert** [comment]
Yeah, the Python library default is also not ideal, leading to similar speed issues with frameworks like Tornado (at least until 4.3 when they [fixed this](http://www.tornadoweb.org/en/stable/releases/v4.3.0.html?highlight=gzip#tornado-web)).  It makes gzip look much worse than it should be in practice.

Set to level 6, gzip was 15% slower than snappy on compression but 40% smaller for the files I was working with.  Decoding with gzip is slower than snappy, of course (although would have been worth it if I was streaming the data over the network at less than 12 MB/sec).

In pyarrow, it looks like they hardcode the gzip compression level to `Z_DEFAULT_COMPRESSION`:

https://github.com/apache/arrow/blob/84e5e02fbf412c979387b0a53b0ad0c6d5c5e790/cpp/src/arrow/util/compression_zlib.cc#L73

which [zlib defines to be 6](https://www.zlib.net/manual.html#Basic).

**Comment by martindurant** [comment]
Happy to accept a PR where the default is level=6. 
Still would try zstd/lz4/lzo or even brotli (whatever that is!) too, if the other end can read them.

**Comment by aiqc** [comment]
It would be nice to mention the reasoning for this here:
https://github.com/dask/fastparquet/blob/efd3fd19a9f0dcf91045c31ff4dbb7cc3ec504f2/fastparquet/compression.py#L14

I was digging around looking for the reason for 6

**Comment by martindurant** [comment]
This was probably the default in hive, parquet-mr or elsewhere. Level 9 is a bad choice anyway, takes ages with minimal benefit. Actually, I would recommend snappy or zstd for quick or good compression, respectively.

**Comment by aiqc** [comment]
I did some testing and snappy compressed files were twice as big as gzip
https://stackoverflow.com/a/64049316/5739514

**Comment by martindurant** [comment]
Yes: snappy is much faster, but not as well compressed. Zstd is about the same speed to compress, as good a compression ratio, and faster to decompress.
